jib-cli/src/main/java/com/google/cloud/tools/jib/cli/war/WarFiles.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.war;
+
+import com.google.cloud.tools.jib.api.InvalidImageReferenceException;
+import com.google.cloud.tools.jib.api.Jib;
+import com.google.cloud.tools.jib.api.JibContainerBuilder;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import com.google.cloud.tools.jib.cli.ArtifactProcessor;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.List;
+
+public class WarFiles {
+
+  /**
+   * Generates a {@link JibContainerBuilder} from contents of a WAR file.
+   *
+   * @param processor artifact processor
+   * @return JibContainerBuilder
+   * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
+   *     provided doesn't exist
+   * @throws InvalidImageReferenceException if the base image reference is invalid
+   */
+  public static JibContainerBuilder toJibContainerBuilder(ArtifactProcessor processor)
+      throws IOException, InvalidImageReferenceException {
+    JibContainerBuilder containerBuilder = Jib.from("jetty");
+    List<FileEntriesLayer> layers = processor.createLayers();
+
+    // JVM Flags are ignored
+    List<String> entrypoint = processor.computeEntrypoint(ImmutableList.of());
```

Comment:
Wait, why doesn't `ArtifactProccesor.computeEntrypoint()` accept `--program-args`? How is `--program-args` being used currently?

Reply:
The program arguments are currently set using `containerBuilder.setProgramArguments()`. But now that I think about it, maybe `computeEntrypoint` in `ArtifactProcessor` shouldn't take in `jvm-flags` as an input. It isn't used in the `War` case at the moment and can cause some confusion in the future.

Reply:
I was thinking we could add `program-args` and `jvm-flags` as parameters to the constructors of the jar processors? to avoid having unused params.

Reply:
Ah, I forgot that `programArguments` corresponds to `CMD` in `Dockerfile`, which is different from `ENTRYPOINT`. So maybe it's okay that `computeEntrypoint` doesn't accept `programArguments`.

Reply:
Ah right. Out of curiosity - since program arguments are added at the end of the entrypoint anyway, both could work right? (i.e. using `setProgramArguments` and making `computeEntrypoint()` accept `programArguments`)

Reply:
I think using `setProgramArguments` is right. Making `computeEntrypoint()` accept it doesn't make sense to me, since `ENTRYPOINT` and `CMD` are two different properties. It's just that I got confused earlier.